### PR TITLE
adding a console command parser

### PR DESCRIPTION
**Pull request by karikittie**
Adding the framework for the console command parser.

### CODE REVIEW

File: FtpClient/FtpCli/ConsoleParser/ConsoleParser.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    public void executeCommand(string command)
+    {
+      string[] args = command.Split(' ');
+      string commandName = args[0];
+      bool found = false;
+      foreach (ConsoleParserCommand c in commands)
+      {
+        if (c.commandName == commandName)
+        {
+          if (args.Length < 2)
```
Comment (mkduer) [inline FtpClient/FtpCli/ConsoleParser/ConsoleParser.cs]: Could there be valid situations where `args.Length == 1`? For example, if we have a command like `ls` it probably wouldn't take additional arguments. Other than that, looks good!

File: FtpClient/FtpCli/ConsoleParser/ConsoleParserCommand.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        public string commandName { get; }
+        private string helpText { get; set; }
+        private Action<List<string>> manyArgsAction;
+        private Action<string> singleArgAction;
+
+        public ConsoleParserCommand(string commandName, string helpText, Action<List<string>> action)
+        {
+            this.commandName = commandName;
+            this.helpText = helpText;
+            this.manyArgsAction = action;
```
Comment (mkduer) [inline FtpClient/FtpCli/ConsoleParser/ConsoleParserCommand.cs:16]: Should `this.manyArgsAction` be initialized and copied into? Example:

`this.manyArgsAction = new List<string>();`
`action.CopyTo(this.manyArgsAction);`

**Review by mkduer — CHANGES_REQUESTED** [review]

**Review by mkduer — DISMISSED** [review]

**Review by bnert — APPROVED** [review]

**Comment by bnert** [comment]
@karikittie, looks good!

**Review by bahamieh — APPROVED** [review]
Cool!

**Pull request merged**